### DISCUSSION TITLE
Extract metadata using a subset of a file

### DISCUSSION THREAD

**Discussion by jorgen-wang** [drewnoakes/metadata-extractor#607] [Q&A] [answered]
Hi,

I'm trying to extract metadata from files stored on azure. It turns out that streams from azure blobs do not support the Seek operation which is used a lot in this library.
As a workaround, if the file is very big, i just download the first MB and save it to a memorystream. Then use the memorystream as input to the library. This seems to work fine for the couple of formats I have tested.
Do you know of any cases where this will blow up in my face? Or is this a valid workaround?

**Comment by drewnoakes** [comment]
Can you use the built-in [`RandomAccessStreamReader`](https://github.com/drewnoakes/metadata-extractor/blob/master/Source/com/drew/lang/RandomAccessStreamReader.java) to do what you're after? It will lazily buffer data in order to allow seeking. In the worst case it'll load all data into memory, but as metadata is often at the start of the file it should avoid doing that much of the time.

**Reply by jorgen-wang** [reply]
That is a great suggestion. I'm using the dotnet version and it doesn't seem to be available there, but I guess I could just make it myself.

Thanks!

**Reply by drewnoakes** [reply, marked as answer]
In the .NET library you are probably looking for [`IndexedCapturingReader`](https://github.com/drewnoakes/metadata-extractor-dotnet/blob/master/MetadataExtractor/IO/IndexedCapturingReader.cs).
